### PR TITLE
MINOR - Prepare App Instances & fix bigtable json

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes:

The main change here is to:
- do not initialize the app multiple times (after every configure, trigger, etc.)
- allow us to pick up initialized instances of applications. This is needed for Collate MetaPilot resource

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/apps/AppException.java
Comment (sushi30) [inline openmetadata-service/src/main/java/org/openmetadata/service/apps/AppException.java:1]: what is this change?

Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/AppException.java:1]: just taking the chance to prepare some named exceptions that I need for another PR in Collate :)

Comment (pmbrull) [reply openmetadata-service/src/main/java/org/openmetadata/service/apps/AppException.java:1]: I'll do a couple of extra additions here. Missing some more functionalities

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/AppRepository.java [hunk 1]
```diff
@@ -164,12 +164,12 @@ public void storeRelationships(App entity) {
...
 
-  public final List<AppRunRecord> listAll() {
+  public final List<App> listAll() {
     // forward scrolling, if after == null then first page is being asked
     List<String> jsons = dao.listAfterWithOffset(Integer.MAX_VALUE, 0);
-    List<AppRunRecord> entities = new ArrayList<>();
+    List<App> entities = new ArrayList<>();
     for (String json : jsons) {
-      AppRunRecord entity = JsonUtils.readValue(json, AppRunRecord.class);
+      App entity = JsonUtils.readValue(json, App.class);
```
Comment (pmbrull) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/AppRepository.java:172]: hi @mohityadav766 this method was not used anywhere, but I think it was casting to the wrong class. Fixed it and using it to init apps on server start

Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/AppRepository.java:172]: that's correct @pmbrull

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**